github.go

```diff
@@ -164,15 +164,37 @@ func getCommits(issueable Issueable, pullRequests PullRequests) ([]*github.Repos
 			message := fmt.Sprintf("Getting commits for PR %s failed", issue.FullName())
 			return nil, &ErrorResponse{err, http.StatusBadGateway, message}
 		}
+		isLastPage := resp.NextPage == 0
+		if isLastPage && !isExpectedHead(*lastCommit(pageCommits).SHA) {
```

Comment:
It might help communicate the background of this condition with a comment (or if possible illuminating variable name).
```go
isCommitListOutdated = isLastPage && !isExpectedHead(*lastCommit(pageCommits).SHA)
if isCommitListOutdated
    ...
```